app/src/test/java/com/example/bottomnav/ParseTest.java

```diff
@@ -0,0 +1,171 @@
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+
+public class ParseTest {
+
+    public class MakeList {
+        public HashMap<String, Contents> repo = new HashMap<>();
+        public MakeList (HashMap<String, Contents> givenRepo){
+            repo = givenRepo;
+        }
+
+        public Contents get(String key) throws IllegalAccessError{
+            if (repo.get(key) == null) {
+                return null;
+            }
+            return repo.get(key);
+        }
+
+        public ArrayList<Contents> getByType(String key) {
+            return findList(key, 0);
+        }
+
+        public ArrayList<Contents> getTypeQualifier(String key) {
+            return findList(key, 1);
+        }
+
+        /**Makes and return a list of contents*/
+        private ArrayList<Contents> findList (String key, int i) {
+            ArrayList<Contents> list = new ArrayList<>();
+            for (String con : repo.keySet()) {
+                Contents contents = get(con);
+                if (i == 0) {
+                    if (contents.type.equals(key)) {
+                        list.add(contents);
+                    }
+                }
+                else {
+                    if (contents.typeQualifer.equals(key)) {
+                        list.add(contents);
+                    }
+                }
+            }
+            if (list.isEmpty()) {
+                throw new IllegalArgumentException("No contents[s] were found for given key.");
+            }
+            return list;
+        }
+    }
+
+    @Test
+    public void doesNotExist() throws Exception {
+        String code = "const uint16_t CAN_ID;";
+        Parse test = new Parse(code);
+
+        assertEquals(null, test.get("CAN_ID"));
+        assertEquals(null, test.get("ducky"));
+    }
+
+    @Test
+    public void simpleCheck() throws Exception {
+        String code = "const uint16_t CAN_ID = 0x16; \n" +
+                "const uint16_t CAN_ID2 = 0x32; \n" +
+                "const uint16_t CAN_ID3 = 0x64;";
+
+        Parse test = new Parse(code);
+
+        HashMap<String, Contents> repository = test.constRepo;
+
+        assertEquals(true, repository.containsKey("CAN_ID"));
+        assertEquals(true, repository.containsKey("CAN_ID2"));
+        assertEquals(true, repository.containsKey("CAN_ID3"));
+
+        assertEquals("const", test.get("CAN_ID").typeQualifer);
+        assertEquals("const", test.get("CAN_ID2").typeQualifer);
+        assertEquals("const", test.get("CAN_ID3").typeQualifer);
+
+        assertEquals("uint16_t", test.get("CAN_ID").type);
+        assertEquals("uint16_t", test.get("CAN_ID2").type);
+        assertEquals("uint16_t", test.get("CAN_ID3").type);
+
+        assertEquals("0x16", test.get("CAN_ID").value);
+        assertEquals("0x32", test.get("CAN_ID2").value);
+        assertEquals("0x64", test.get("CAN_ID3").value);
+    }
+
+    @Test
+    public void checkdiverse() throws Exception {
```

Comment:
Consider a more specific / descriptive name that gets at what this test is checking for - perhaps it's the different types?